Tests/store_packs.py

```diff
@@ -0,0 +1,138 @@
+import os
+import sys
+import argparse
+from zipfile import ZipFile, ZIP_DEFLATED
+from Tests.test_utils import run_command, print_error
+from google.cloud import storage
+import google.auth
+
+
+def option_handler():
+    parser = argparse.ArgumentParser(description="Store packs in cloud storage.")
+    parser.add_argument('-a', '--artifactsPath', help="The full path of packs artifacts", required=True)
+    parser.add_argument('-e', '--extractPath', help="Full path of folder to extract wanted packs", required=True)
+    parser.add_argument('-c', '--circleCi', help="Whether run script locally or in circleCi", default=False)
+    parser.add_argument('-p', '--packName', help="Use only in local mode, the target pack name to store.",
+                        required=False, default="")
+    parser.add_argument('-b', '--bucketName', help="Storage bucket name", required=True)
+
+    return parser.parse_args()
+
+
+class Pack:
+    CHANGELOG = "changelog.json"
+    PACK_INITIAL_VERSION = "1.0.0"
+
+    def __init__(self, pack_name, pack_path, storage_bucket):
+        self.pack_name = pack_name
+        self.pack_path = pack_path
+        self.storage_bucket = storage_bucket
+
+    @property
+    def latest_version(self):
+        return self._get_latest_version()
+
+    def _get_latest_version(self):
+        if self.CHANGELOG not in os.listdir(self.pack_path):
+            return self.PACK_INITIAL_VERSION
+
+        changelog_path = os.path.join(self.pack_path, self.CHANGELOG)
+
+        with open(changelog_path, "r") as changelog:
+            changelog_json = json.load(changelog)
+
+            pack_versions = list(changelog_json.keys())
+            pack_versions.sort(key=lambda str_version: [int(v) for v in str_version.split(".")],
+                               reverse=True)
+
+            return pack_versions[0]
+
+    def zip_pack(self):
+        zip_pack_path = f"{self.pack_path}.zip"
+
+        with ZipFile(zip_pack_path, 'w', ZIP_DEFLATED) as pack_zip:
+            for root, dirs, files in os.walk(self.pack_path):
+                for f in files:
+                    full_file_path = os.path.join(root, f)
+                    relative_file_path = os.path.relpath(full_file_path, self.pack_path)
+                    pack_zip.write(filename=full_file_path, arcname=relative_file_path)
+
+        return zip_pack_path
+
+    def upload_to_storage(self, zip_pack_path, latest_version):
+        version_pack_path = f"content/packs/{self.pack_name}/{latest_version}"
+        existing_files = [f.name for f in self.storage_bucket.list_blobs(prefix=version_pack_path)]
+
+        if len(existing_files) > 0:
+            print_error(f"The following packs already exist at storage: {', '.join(existing_files)}")
+            print_error(f"Skipping step of uploading {self.pack_name}.zip to storage.")
+            sys.exit(1)
+
+        pack_full_path = f"{version_pack_path}/{self.pack_name}.zip"
+        blob = self.storage_bucket.blob(pack_full_path)
```

Comment:
blob?

Reply:
Official google storage python client library, they call the object stored in the path blobs